src/SAWScript/Interpreter.hs

```diff
@@ -1913,6 +1921,15 @@ primitives = Map.fromList
     , "The specified size must be greater than the size of the LLVM type."
     ]
 
+  , prim "crucible_symbolic_alloc" "Bool -> Int -> Term -> CrucibleSetup SetupValue"
+    (bicVal crucible_symbolic_alloc)
+    Current
+    [ "Like `crucible_alloc`, but with a (symbolic) size instead of"
+    , "a LLVM type. The first argument specifies whether the allocation is"
+    , "read-only. The second argument specifies the alignment in bytes (which"
```

Comment:
This might be more of a SAW-wide discussion (or maybe it's something we're doing in Python) but it would be great to move away from booleans and into a datatype that actually describes what it's doing. e.g. ReadOnly or ReadWrite in this case

Reply:
cc @atomb

Reply:
Yeah, the Python interface works more like this. The type system of SAWScript makes it a bit more of a pain to do that. It doesn't have algebraic data types, so the best you can do is introduce some abstract type and various individual constants that also have that type. It's possible, but awkward.